### PR TITLE
Boxcar wave fix

### PR DESCRIPTION
**Pull request by jhennawi**
This PR fixes a bug associated with 2d interpolation of waveimg in extraction.

### CODE REVIEW

File: pypeit/core/extract.py [hunk 1]
```diff
@@ -175,10 +175,16 @@ def extract_optimal(sciimg, ivar, mask, waveimg, skyimg, thismask, oprof,
                                    np.nansum(thismask_sub[oprof_good,:]*oprof_sub[oprof_good,:]**2, axis=1)
         oprof_bad = badwvs & ((oprof_smash <= 0.0) | (np.isfinite(oprof_smash) == False) | (wave_opt <= 0.0) | (np.isfinite(wave_opt) == False))
         if oprof_bad.any():
+            # If there are no good profile wavelengths, use boxcar wavelengths for these pixels
+            # get boxcar_radius
+            box_radius = spec.BOX_RADIUS
```
Comment (profxj) [inline pypeit/core/extract.py:180]: Can you explain why the change is needed and effective?
  Thx.

File: pypeit/core/extract.py [hunk 2]
```diff
@@ -175,10 +175,16 @@ def extract_optimal(sciimg, ivar, mask, waveimg, skyimg, thismask, oprof,
...
+            box_denom_no_mask = moment1d(waveimg > 0.0, spec.TRACE_SPAT, 2 * box_radius, row=spec.trace_spec)[0]
+            wave_no_mask = moment1d(waveimg, spec.TRACE_SPAT, 2 * box_radius, row=spec.trace_spec)[0] / (
+                        box_denom_no_mask + (box_denom_no_mask == 0.0))
+            wave_opt[oprof_bad] = wave_no_mask[oprof_bad]
             # For pixels with completely bad profile values, interpolate from trace.
-            f_wave = scipy.interpolate.RectBivariateSpline(spec_vec,spat_vec, waveimg*thismask)
-            wave_opt[oprof_bad] = f_wave(spec.trace_spec[oprof_bad], spec.TRACE_SPAT[oprof_bad],
-                                         grid=False)
+            #f_wave = scipy.interpolate.RectBivariateSpline(spec_vec,spat_vec, waveimg*thismask)
```
Comment (profxj) [inline pypeit/core/extract.py:186]: Remove?

**Review by profxj — CHANGES_REQUESTED** [review]
Only requesting an explanation for the change

And I'm thinking a full run of tests is in order..

File: pypeit/core/extract.py (hunk 1 above)
Comment (jhennawi) [reply pypeit/core/extract.py:180]: The interpolation is problematic if lots of pixels get masked. The current fix is more robust.

**Comment by profxj** [comment]
@badpandabear -- Can you please launch a Cloud test run on this branch?
  Thanks

**Comment by badpandabear** [comment]
@profxj -- Dev suite started

File: pypeit/core/extract.py (hunk 2 above)
Comment (jhennawi) [reply pypeit/core/extract.py:186]: Let's leave the old code there for now in case we need to revert.

**Review by profxj — APPROVED** [review]
ready to go

**Review by debora-pe — APPROVED** [review]
Looks good to me too.
But, please, post the result of the tests.

**Comment by badpandabear** [comment]
Testing first run:
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  200 passed, 1195 warnings in 145.76s (0:02:25) ---
--- PYTEST UNIT TESTS PASSED  115 passed, 1803 warnings in 750.03s (0:12:30) ---
--- PYTEST VET TESTS PASSED  24 passed, 1566 warnings in 719.73s (0:11:59) ---
--- PYPEIT DEVELOPMENT SUITE FAILED 2/151 TESTS  ---
Failed tests:
    keck_deimos/830G_M_9000_dither pypeit 
    ldt_deveny/DV1 pypeit 
Skipped tests:
    keck_deimos/830G_M_9000_dither pypeit_coadd_2dspec
Testing Started at 2022-08-09T16:52:29.980706
Testing Completed at 2022-08-09T23:26:55.369954
Total Time: 6:34:25.389248
```
After retrying:
```
Running tests on the following instruments:
    keck_deimos
    ldt_deveny

Reducing data from keck_deimos for the following setups:
    830G_M_9000_dither

Reducing data from ldt_deveny for the following setups:
    DV1

Running tests in 6 parallel processes
 0 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_9000_dither pypeit 
 0 passed/ 0 failed/ 0 skipped STARTED ldt_deveny/DV1 pypeit 
 1 passed/ 0 failed/ 0 skipped PASSED  ldt_deveny/DV1 pypeit 
 2 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_9000_dither pypeit 
 2 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_9000_dither pypeit_coadd_2dspec
 3 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_9000_dither pypeit_coadd_2dspec
--- PYPEIT DEVELOPMENT SUITE PASSED 3/3 TESTS  ---

Test Summary
--------------------------------------------------------
--- PYPEIT DEVELOPMENT SUITE PASSED 3/3 TESTS  ---
Testing Started at 2022-08-09T23:34:16.765722
Testing Completed at 2022-08-10T01:21:41.526674
Total Time: 1:47:24.760952
```
